File: product-webapp-service/webapp/src/app/model/ProductHistory.ts
import { User } from "./User";
import { Product } from "./product";

export class ProductHistory{
    transaction_id:String;
    rentedProduct:Product
    customer:User;
    orderDate:string;
    deliveryDate:string;
    paymentId:string

    constructor(){
        this.transaction_id=''
        this.rentedProduct=new Product();
        this.customer=new User();
        this.orderDate='';
        this.deliveryDate='';
        this.paymentId='';
    }
}

File: product-history/src/main/java/com/stackroute/producthistory/model/ProductHistory.java
package com.stackroute.producthistory.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.util.Date;
import java.util.UUID;

@Document(collection = "history")
@Getter
@Setter
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ProductHistory {
    @Id
    String transaction_id= UUID.randomUUID().toString();
    Product rentedProduct;
    User customer;
    String paymentId;
    LocalDate orderDate= LocalDate.now();
    LocalDate deliveryDate= orderDate.plusDays(10);
}
